### ISSUE TITLE
Support zero-shot task splits

### ISSUE THREAD

**Issue by fteufel** [polaris-hub/polaris#131]
There are prediction problems that are often evaluated in a zero-shot scenario, with no data used for training. All data would be considered test data, and performance typically evaluated as a correlation efficient between the true values and a score from the model (e.g. perplexity).

Right now, polaris requires that there are at least two non-empty partitions specified in the `split` for `SingleTaskBenchmarkSpecification`. Could this be made more flexible, or is this out of scope for polaris?

Thanks!

**Comment by cwognum** [comment]
Hey @fteufel,

Thanks a lot for the suggestion! It's an interesting proposal and I see how that would be useful. Technically, it shouldn't be too big a challenge either.

It does raise some more "philosophical" questions. One of the main goals we've set out to solve with Polaris is to make sure results are comparable. We want to be comparing methods and e.g. not datasets or preprocessing workflows. If we don't impose a fixed train set, it's not clear to me how we can make sure results remain comparable. For example, a method that is pre-trained on more data or data that comes from the same chemical space as the test set is expected to do better, but is that a meaningful difference _between methods_? Probably not? I'm not super familiar with the one-shot literature. Do you know if this is something that is addressed in a paper you could reference for me to read?

Or given the above, maybe we should not allow test-set only datasets on Polaris? Instead, we could ask people in a zero-shot setting to always explicitly define a possibly unsupervised (pre)train set? What do you think? In that case though, maybe we can think of other features that make defining such a benchmark easier, e.g. by allowing people to specify benchmarks based on two or more datasets?

Thanks again for the suggestion! Let me know what you think of this.

**Comment by cwognum** [comment]
> people might use pretrained representations to build their supervised model on top of

That's a great point!

I think you're right that there is no way to fully prevent this and that may indeed not be as big a problem.

I do still think we should at some
point address the pre-training case. Maybe we could e.g. add a metadata field in the results object that lets you specify the precise data you pretrained on. Simply being transparent about this seems useful in comparing methods.

Thanks, @fteufel ! These are all interesting points!

**Comment by fteufel** [comment]
That makes a lot of sense - definitely would be valuable to have pretraining info available. 

So, can we have zero-shot? 🙂

**Comment by cwognum** [comment]
Yes! Do you want to make a PR?
